src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -793,8 +799,83 @@ internal List<INodeLibraryDependencyInfo> NodeLocalDefinitions
             }
         }
 
+        internal List<INodeLibraryDependencyInfo> ExternalFiles
+        {
+            get
+            {
+                return GetExternalFiles();
+            }
+            set
+            {
+                foreach (var dependency in value)
+                {
+                    if (dependency.ReferenceType == ReferenceType.External)
+                    {
+                        foreach (var node in dependency.Nodes)
+                        {
+                            externalFilesDictionary[node] = dependency as DependencyInfo;
+                        }
+                    }
+                }
+
+                RaisePropertyChanged(nameof(ExternalFiles));
+            }
+        }
+
+        private List<INodeLibraryDependencyInfo> GetExternalFiles()
+        {
+            var externalFiles = new Dictionary<object, DependencyInfo>();
+
+            // Computes the external file references if the Workspace Model is a HomeWorkspaceModel.
+            // The workspace should be executed for the external references to be computed because the node output port values are needed.
+            if (this is HomeWorkspaceModel homeWorkspaceModel)
+            {
+                foreach (var node in nodes)
+                {
+                    externalFilesDictionary.TryGetValue(node.GUID, out var serializedDependencyInfo);
+
+                    // Check for the file path string value at each of the output ports of all nodes in the workspace. 
+                    foreach (var port in node.OutPorts)
+                    {
+                        var id = node.GetAstIdentifierForOutputIndex(port.Index).Name;
+                        var mirror = homeWorkspaceModel.EngineController.GetMirror(id);
+                        var data = mirror?.GetData().Data;
+
+                        if (data is string dataString)
+                        {
+                            // If the value exists on disk
+                            if (File.Exists(data.ToString()))
```

Comment:
Can you check if there is a quicker way to check for path string before making system calls? I suspect this may introduce performance issue with large graph. If there is no good way, should we check for string including `\` as a first path?

Reply:
have you looked at FileInfo constructor instead? 
https://stackoverflow.com/questions/3067479/determine-via-c-sharp-whether-a-string-is-a-valid-file-path